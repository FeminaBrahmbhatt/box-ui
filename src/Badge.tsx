import React, { FC } from "react";
import { BadgeProps, BadgeSize, BadgeVariant } from "./@interfaces";
import classNames from "classnames";

const BadgeSizeClasses: Record<BadgeSize, string> = {
  sm: "badge-sm",
  md: "badge-md",
  lg: "badge-lg",
};

const BadgeVariantClasses: Record<BadgeVariant, string> = {
  gray: "badge-gray",
  primary: "badge-primary",
  warning: "badge-warning",
  error: "badge-error",
  success: "badge-success",
};

export const Badge: FC<BadgeProps> = ({
  className,
  children,
  LeadingIcon,
  TrailingIcon,
  variant,
  size,
}) => {
  const badgeVariantClassName = BadgeVariantClasses[variant];
  const badgeSizeClassName = BadgeSizeClasses[size];

  return (
    <div
      className={classNames(
        "badge-base",
        badgeSizeClassName,
        badgeVariantClassName,
        className
      )}
    >
      {LeadingIcon && <span className="mr-1.5">{LeadingIcon}</span>}
      {children}
      {TrailingIcon && <span className="mr-1.5">{TrailingIcon}</span>}
    </div>
  );
};
