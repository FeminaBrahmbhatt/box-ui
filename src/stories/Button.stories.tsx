import React from "react";
import type { Meta, StoryObj } from "@storybook/react";
import { Button } from "../Button";
import { ButtonProps } from "../@interfaces";
import { FiArrowLeft, FiArrowRight, FiStar } from "react-icons/fi";
import { StoryLayout } from "../StoryLayout";
import { IconWrapper } from "../utils/IconWrapper";

export default{
  title: "Example/Button",
  component: Button,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
} as Meta;
interface Props extends ButtonProps {
  darkMode: boolean;
}

type Story = StoryObj<Props>;

export const Default: Story = {
  args: {
    darkMode: false,
    children: "Button",
    size: 'md',
    variant: 'primary',
    disabled: false,
    LeadingIcon: undefined,
    TrailingIcon: undefined,
    IconOnly: undefined,
    className: ''
  },
  render: (args) => {
    return (
      <StoryLayout {...args} className="flex inline-flex flex-row space-x-2">
        <Button {...args}>Primary</Button>
      </StoryLayout>
    );
  },
};

export const Variants: Story = {
  args: {
    darkMode: false,
    variant: "primary",
  },
  render: (args) => {
    return (
      <StoryLayout {...args} className="flex inline-flex flex-row space-x-2">
        <Button variant="primary">Primary</Button>
        <Button variant="secondary">Secondary</Button>
        <Button variant="secondaryGray">Secondary Gray</Button>
        <Button variant="tertiary">Tertiary</Button>
        <Button variant="tertiaryGray">Tertiary Gray</Button>
      </StoryLayout>
    );
  },
};

export const Sizes: Story = {
  args: {
    darkMode: false,
    variant: "primary",
    size: "md",
  },
  render: (args) => {
    return (
      <StoryLayout {...args} className="flex inline-flex flex-row space-x-2">
        <Button size="sm">sm</Button>
        <Button size="md">md</Button>
        <Button size="lg">lg</Button>
        <Button size="xl">xl</Button>
        <Button size="2xl">2xl</Button>
      </StoryLayout>
    );
  },
};

export const TrailingIconButton: Story = {
  args: {
    darkMode: false,
    children: "Button",
    TrailingIcon: <IconWrapper Icon={FiArrowRight} data-testid="trailingIcon"/>,
  },  
  render: (args) => (
    <StoryLayout {...args}>
      <Button {...args} />
    </StoryLayout>
  ),
};

export const LeadingIconButton: Story = {
  args: {
    darkMode: false,
    children: "Button",
    LeadingIcon: <IconWrapper Icon={FiArrowLeft} data-testid="LeadingIcon"/>,
  },
  render: (args) => (
    <StoryLayout {...args}>
      <Button {...args} />
    </StoryLayout>
  ),
};

export const IconOnlyButton: Story = {
  args: {
    darkMode: false,
    IconOnly: <IconWrapper Icon={FiStar} data-testid="iconOnly"/>,
  },
  render: (args) => (
    <StoryLayout {...args}>
      <Button {...args} />
    </StoryLayout>
  ),
};
