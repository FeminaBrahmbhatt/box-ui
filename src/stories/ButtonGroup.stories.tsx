import React, { useState } from "react";
import { Meta, StoryObj } from "@storybook/react";
import { ButtonGroupProps, ButtonGroup } from "../ButtonGroup";
import { StoryLayout } from "../StoryLayout";
import { options1, options2, options3 } from "../data/options";

export default {
  title: "Example/ButtonGroup",
  component: ButtonGroup,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} as Meta;
interface Props extends ButtonGroupProps {
  darkMode: boolean;
}

const LabelButtonGroupComponent = (args: Props) => {
  const [activeItem1, setActiveItem1] = useState<string>(options1[1].value);

  return (
    <StoryLayout {...args}>
      <ButtonGroup
        {...args}
        options={options1}
        activeOption={activeItem1}
        setActiveOption={setActiveItem1}
      />
    </StoryLayout>
  );
};

export const LabelButtonGroup: StoryObj<Props> = {
  args: {
    darkMode: false,
  },
  render: (args) =>  <LabelButtonGroupComponent {...args} />,
};

const IconButtonGroupComponent = (args: Props) => {
  const [activeView, setViewOption] = useState<string>("list");

  return (
    <StoryLayout {...args}>
      <ButtonGroup
        {...args}
        options={options3}
        activeOption={activeView}
        setActiveOption={setViewOption}
      />
    </StoryLayout>
  );
};

export const IconButtonGroup: StoryObj<Props> = {
  args: {
    darkMode: false,
  },
  render: (args) => <IconButtonGroupComponent {...args} />,
};

const LabelIconButtonGroupComponent = (args: Props) => {
  const [activeItem2, setActiveItem2] = useState<string>(options2[1].value);

  return (
    <StoryLayout {...args}>
      <ButtonGroup
        {...args}
        options={options2}
        activeOption={activeItem2}
        setActiveOption={setActiveItem2}
      />
    </StoryLayout>
  );
};

export const LabelIconButtonGroup: StoryObj<Props> = {
  args: {
    darkMode: false,
  },
  render: (args) => <LabelIconButtonGroupComponent {...args} />,
};
