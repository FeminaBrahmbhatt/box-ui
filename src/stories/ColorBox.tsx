import React from "react";
import { IColor } from "./@interfaces";

export const ColorBox = ({ code, hex, bgClass }: IColor) => (
<div
    key={bgClass}
    className="flex flex-col rounded-lg shadow-lg w-26 h-39 dark:bg-white">
      <div className={`h-20 ${bgClass} rounded-t-lg`} />
      <div className="mx-3 my-3">
          {code}
          <br/>
          {hex}
      </div>
    </div>
);