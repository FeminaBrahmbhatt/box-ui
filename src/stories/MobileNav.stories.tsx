import React, { useState } from "react";
import { Meta, StoryObj } from "@storybook/react";
import { MobileNavProps } from "../@interfaces";
import { SideNav } from "../Nav/SideNav";
import { StoryLayout } from "../StoryLayout";
import { navItemsTop, navItemsBottom } from "../data/navItems";
import { MobileNavbar } from "../Nav/MobileNav";
import { FiX } from "react-icons/fi"; // Directly import FiX
import type { IconBaseProps } from "react-icons";

const CloseIcon = FiX as React.FC<IconBaseProps>;
// Wrapper to fix TS JSX issues
 
export default {
  title: "Example/MobileNav",
  component: MobileNavbar,
} as Meta;

interface Props extends MobileNavProps {
  darkMode: boolean;
}

const MobileNavWithState = (args: Props) => {
  const [open, setOpen] = useState<boolean>(false);

  const handleToggle = () => {
    setOpen(!open);
  };

  return (
    <StoryLayout {...args}>
      <MobileNavbar open={open} toggleOpen={handleToggle} />

      {open && (
        <div
          data-testid="backdrop"
          className="fixed top-0 left-0 z-40 w-screen h-screen bg-gray-500 cursor-pointer bg-opacity-80"
          onClick={() => setOpen(false)}
        >
           <CloseIcon
            data-testid="close-icon"
            size={40}
            className="fixed text-white top-5 right-9"
            role="button"
            aria-label="Close sidebar"
          />
        </div>
      )}

      {open && (
        <SideNav
          className="relative z-50 h-screen -mt-20"
          navItemsTop={navItemsTop}
          navItemsBottom={navItemsBottom}
          open={true}
          setOpen={handleToggle}
          username="Veronica Woods"
          email="veronica@example.com"
        />
      )}
    </StoryLayout>
  );
};

export const Default: StoryObj<Props> = {
  args: {
    darkMode: false,
  },
  render: (args) => <MobileNavWithState {...args} />,
};
