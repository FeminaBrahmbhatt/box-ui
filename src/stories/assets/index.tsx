
export type CountryISO = "BE" | "DE" | "FR" | "IT" | "NL" | "RU" | "US";

type image = CountryISO;

// Development images
const devImages: Record<string, image> = {
  NL: require("./nl.svg")
};

// Production images (CDN)
// const prodImages: Record<string, image> = {
//   NL: "https://res.cloudinary.com/tailwindcss/image/upload/v1635279277/nl_tpy2ab.svg",
// };

export const assets = devImages;