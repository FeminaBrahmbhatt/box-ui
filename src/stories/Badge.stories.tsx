import React from 'react';
import {BadgeProps} from '../@interfaces';
import { Badge } from '../Badge';
import {images} from '../data/images'
import {Meta, StoryObj} from '@storybook/react';
import { StoryLayout } from '../StoryLayout';
import { FiArrowRight, FiStar } from 'react-icons/fi';
import { IconWrapper } from '../utils/IconWrapper';

export default {
    title: 'Example/Badge',
    component: Badge,
    parameters: {
        layout: 'centered',
    },
    tags: ['autodocs'],
} as Meta;
interface Props extends BadgeProps {
    darkMode: boolean;
}

export const Default:StoryObj<Props> = {
    args: {
        darkMode: false,
        variant: "success",
        size: "md",
        children:<span>Label</span>
    }
}

export const LeadingIconBadge:StoryObj<Props>  = {
    args: {
        darkMode: false,
        variant: "success",
        size: "md",
        LeadingIcon: <IconWrapper Icon={FiStar} data-testid="leadingIcon"/>
    },
    render: (args) => {
        return (
            <StoryLayout {...args} className="inline-flex flex-col space-y-2">
                <Badge {...args}>
                    Label
                </Badge>
            </StoryLayout>
        )
    }
}

export const ImageIconBadge:StoryObj<Props>  = {
    args: {
        darkMode: false,
        variant: "gray",
        size: "md",
        LeadingIcon: <img src={images.NL} alt="nl" className="w-4 h-4" data-testid="iconOnly" />
    },
    render: (args) => {
        return (
            <StoryLayout {...args} className="inline-flex flex-col space-y-2">
                <Badge {...args}>
                    Newzeland
                </Badge>
            </StoryLayout>
        )
    }
}

export const TrailingIconBadge:StoryObj<Props>  = {
    args: {
        darkMode: false,
        variant: "primary",
        size: "md",
        TrailingIcon: <IconWrapper Icon={FiArrowRight} data-testid="trailingIcon"/>
    },
    render: (args) => {
        return (
            <StoryLayout {...args} className="inline-flex flex-col space-y-2">
                <Badge {...args}>
                    Label
                </Badge>
            </StoryLayout>
        )
    }
}

