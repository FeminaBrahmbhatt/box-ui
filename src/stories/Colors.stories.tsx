import React from "react";
import { Meta, StoryObj } from "@storybook/react";
import { ColorBox } from "../ColorBox";
import { colors } from "../data/colors";
import { StoryLayout } from "../StoryLayout";

export default {
  title: "Colors",
  parameters: {
    controls: { expanded: true },
  },
} as Meta;

interface Props {
  darkMode: boolean;
}

export const Default: StoryObj<Props> = {
  args: {
    darkMode: false,
  },
  render: (args) => (
    <StoryLayout {...args} className="grid grid-cols-2 gap-4 md:grid-cols-4 lg:grid-cols-6 xl:grid-cols-11">
      {colors.map((color) => (
        <ColorBox key={color.bgClass} code={color.code} hex={color.hex} bgClass={color.bgClass}/>
      ))}
    </StoryLayout>
  ),
};
