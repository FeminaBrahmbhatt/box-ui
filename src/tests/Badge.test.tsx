
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { composeStories } from '@storybook/react';
import * as stories from '../stories/Badge.stories';

const {
    Default, LeadingIconBadge, ImageIconBadge, TrailingIconBadge
  } = composeStories(stories);


describe('ðŸ“˜ Badge Storybook Stories', () => {
  test('renders Default badge', () => {
    const Story = Default;
    const { container } = render(
        Story(Story.args ?? {})
      );
    expect(screen.getByText('Label')).toBeInTheDocument();
    expect(container).toMatchSnapshot(); // Optional: snapshot for structure
  });

  test('renders Badge with leading icon', () => {
    const Story = LeadingIconBadge;
    render(
        Story(Story.args ?? {})
      );
    expect(screen.getByText('Label')).toBeInTheDocument();
    expect(screen.getByTestId('leadingIcon')).toBeInTheDocument(); // Checks for <FiStar /> icon
  });

  test('renders Badge with image icon', () => {
    const Story = ImageIconBadge;
    render(
        Story(Story.args ?? {})
      );
    expect(screen.getByText('Newzeland')).toBeInTheDocument();
    expect(screen.getByTestId('iconOnly')).toBeInTheDocument(); // Checks for image icon
  });

  test('renders Badge with trailing icon', () => {
    const Story = TrailingIconBadge;
    render(
        Story(Story.args ?? {})
      );
    expect(screen.getByText('Label')).toBeInTheDocument();
    expect(screen.getByTestId('trailingIcon')).toBeInTheDocument(); // Checks for <FiArrowRight /> icon
  });
});
