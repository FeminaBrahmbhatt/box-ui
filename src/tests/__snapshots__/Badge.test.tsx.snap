// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ“˜ Badge Storybook Stories renders Default badge 1`] = `
<div>
  <div
    class="badge-base badge-md badge-success"
  >
    <span>
      Label
    </span>
  </div>
</div>
`;
